app/controllers/users_controller.rb

```diff
@@ -26,8 +26,9 @@ def create
 
   # PATCH/PUT /users/1
   def update
-    if @user.update(user_params)
-      render json: @user
+    @user = User.joins(:books).find(params[:id])
```

Comment:
this is an odd way of doing this books action. controllers for user should only look for users. here you are looking for the transaction id since you are looking for the joins first. i know this was old code but now we go to the transaction controller first, so clean it up.